code/modules/jobs/job_types/security_officer.dm

```diff
@@ -161,19 +161,23 @@ GLOBAL_LIST_EMPTY(security_officer_distribution)
 		partners += partner.real_name
 
 	if (partners.len)
-		for (var/obj/item/modular_computer/pda as anything in GLOB.TabletMessengers)
-			if (pda.saved_identification in partners)
-				targets += pda
+		for(var/msgr_ref in GLOB.TabletMessengers)
+			var/datum/computer_file/program/messenger/msgr = GLOB.TabletMessengers[msgr_ref]
+			if(!(msgr.computer?.saved_identification in partners))
+				continue
+			targets += msgr
 
 	if (!targets.len)
 		return
 
+	var/datum/pda_msg/msg = new("Officer [officer.real_name] has been assigned to your department, [department].", TRUE)
+
 	var/datum/signal/subspace/messaging/tablet_msg/signal = new(announcement_system, list(
```

Comment:
Why are we keeping this over using ``/datum/pda_msg`` directly?

Reply:
`/datum/pda_msg` is a simple data structure meant for logging messages inside the messenger and can't be used to transmit signals. However, I am unhappy about passing instances of it around for signal transmission and I'm thinking about dissolving the fields into the signal data itself.